models/experimental/yolov4/demo/demo_yolov4_2cq_trace.py

```diff
@@ -0,0 +1,714 @@
+
+            boxes = post_processing(img, conf_thresh, nms_thresh, output)
+
+            class_names = load_class_names(class_name)
+            img = cv2.imread(imgfile)
+
+            plot_boxes_cv2(img, boxes[0], "ttnn_prediction_demo.jpg", class_names)
+
+        else:
+            t1 = time.time()
+            output = model(img)
+
+            yolo1 = YoloLayer(
+                anchor_mask=[0, 1, 2],
+                num_classes=n_classes,
+                anchors=[12, 16, 19, 36, 40, 28, 36, 75, 76, 55, 72, 146, 142, 110, 192, 243, 459, 401],
+                num_anchors=9,
+                stride=8,
+            )
+
+            yolo2 = YoloLayer(
+                anchor_mask=[3, 4, 5],
+                num_classes=n_classes,
+                anchors=[12, 16, 19, 36, 40, 28, 36, 75, 76, 55, 72, 146, 142, 110, 192, 243, 459, 401],
+                num_anchors=9,
+                stride=16,
+            )
+
+            yolo3 = YoloLayer(
+                anchor_mask=[6, 7, 8],
+                num_classes=n_classes,
+                anchors=[12, 16, 19, 36, 40, 28, 36, 75, 76, 55, 72, 146, 142, 110, 192, 243, 459, 401],
+                num_anchors=9,
+                stride=32,
+            )
+
+            y1 = yolo1(output[0])
+            y2 = yolo2(output[1])
+            y3 = yolo3(output[2])
+
+            output = get_region_boxes([y1, y2, y3])
+
+            t2 = time.time()
+
+            print("-----------------------------------")
+            print("           Preprocess : %f" % (t1 - t0))
+            print("      Model Inference : %f" % (t2 - t1))
+            print("-----------------------------------")
+
+            boxes = post_processing(img, conf_thresh, nms_thresh, output)
+
+            class_names = load_class_names(class_name)
+            img = cv2.imread(imgfile)
+            plot_boxes_cv2(img, boxes[0], "torch_prediction_demo.jpg", class_names)
+
+
+####
+
+
+def run_trace_2cq_model(device, test_infra, num_warmup_iterations, num_measurement_iterations):
+    ops_parallel_config = {}
+    tt_inputs_host, sharded_mem_config_DRAM, input_mem_config = test_infra.setup_dram_sharded_input(device)
+    tt_image_res = tt_inputs_host.to(device, sharded_mem_config_DRAM)
+
+    op_event = ttnn.create_event(device)
+    write_event = ttnn.create_event(device)
+    # Initialize the op event so we can write
+    ttnn.record_event(0, op_event)
+
+    profiler.start("compile")
+    ttnn.wait_for_event(1, op_event)
+    ttnn.copy_host_to_device_tensor(tt_inputs_host, tt_image_res, 1)
+    ttnn.record_event(1, write_event)
+    ttnn.wait_for_event(0, write_event)
+    test_infra.input_tensor = ttnn.to_memory_config(tt_image_res, input_mem_config)
+    shape = test_infra.input_tensor.shape
+    dtype = test_infra.input_tensor.dtype
+    layout = test_infra.input_tensor.layout
+    ttnn.record_event(0, op_event)
+    # _ = ttnn.from_device(test_infra.run(), blocking=True)
+    _ = test_infra.run()
```

Comment:
We don't include readback time in the compile/cache runs?
Does `test_infra.run()` include a synchronize/blocking call? Otherwise we should have one here if we're not doing readback in order to properly measure the time

Reply:
@tt-aho this was copied from RN50 code, but I had to commit the ttnn.from_device()
The `test_infra.run()` is as simple as :
```
def run(self, tt_input_tensor=None):
        self.output_tensor = None
        self.output_tensor = self.ttnn_model(self.device, self.input_tensor)
        return self.output_tensor

```

Reply:
The issue I have 3 tensors output, so do I need 3 or 1 blocking calls? please advise what's the better approach here. 
<img width="1690" alt="image" src="https://github.com/user-attachments/assets/4f6df112-47b4-49a1-8728-6f15d03e3ba2">